### PR TITLE
Add atoti recipe

### PR DESCRIPTION
**Pull request by fabiencelier**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by rcolle** [comment]
I'm willing to be listed as a mainter.

**Comment by tibdex** [comment]
I'm willing to be listed as a maintainer.

**Comment by tibdex** [comment]
> The recipe `license` should not include the word "License".

This should now be allowed since https://github.com/conda-forge/conda-smithy/commit/692ab8d39993be7f82a0a93155e07eb92fdf7b37.

**Comment by fabiencelier** [comment]
@conda-forge/help-python Can you review this PR ?

File: recipes/atoti/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  sha256: a7c4793c1a082091c88d6ef3760f714add65ea39f1998a0fe5ce063aec3c271e
+
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} setup.py install --single-version-externally-managed --record=record.txt"
+
+requirements:
+  build:
```
Comment (beckermr) [inline recipes/atoti/meta.yaml:18]: This should be host not build.

File: recipes/atoti/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+source:
+  url: https://github.com/atoti/atoti/releases/download/v{{ version }}/atoti-{{ version }}.tar.gz
+  sha256: a7c4793c1a082091c88d6ef3760f714add65ea39f1998a0fe5ce063aec3c271e
+
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} setup.py install --single-version-externally-managed --record=record.txt"
```
Comment (beckermr) [inline recipes/atoti/meta.yaml:15]: Use pip. See the example recipe.

File: recipes/atoti/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+requirements:
+  build:
+    - python =3.7
```
Comment (beckermr) [inline recipes/atoti/meta.yaml:19]: This should be >=

File: recipes/atoti/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    - setuptools
```
Comment (beckermr) [inline recipes/atoti/meta.yaml:20]: This dep should be pip

File: recipes/atoti/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+
+test:
+  requires:
+    - openjdk
+    - pandas
+
+about:
+  home: https://www.atoti.io
+  license: https://www.atoti.io/eula/
```
Comment (beckermr) [inline recipes/atoti/meta.yaml:35]: This needs to be the kind of license per spdx not a link

Comment (fabiencelier) [reply recipes/atoti/meta.yaml:35]: Hello @beckermr 
Thank you for your review and your help.
This is a custom license so I don't know which kind of SPDX I should use. Is there a "Custom" option or something ? What should I do ?

Comment (beckermr) [reply recipes/atoti/meta.yaml:35]: You should use a LicenseRef expression. @isuruf can help with which one.

**Comment by jbe456** [comment]
Hello @isuruf! Thanks for taking the time to look at this PR. Could you provide more information regarding your last statement? What is the issue with the EULA and what needs to be changed?

**Comment by beckermr** [comment]
Good catch @isuruf. I agree with your assessment. Distribution on conda-forge requires more BSD or GNU style licenses that give the user the right to distribute the software in any way they like subject to the typical conditions.

**Comment by isuruf** [comment]
@jbe456, the package doesn't have to be open-source, but the EULA needs to allow redistribution.  Also, the following statement is problematic as well.

> 6.1. This EULA is effective for a period of 64 (sixty-four) days from the date of the Software’s release (the “Term”). At the end of the Term, the EULA is automatically terminated and End User is not authorized to use the Software any longer.

**Comment by jbe456** [comment]
Thanks I understand now. Do you confirm these are the only two issues? We'll update the EULA and come back to you.

About the 64 days period, the idea was to push people to upgrade to more recent versions. We could rephrase it in that sense if that's still acceptable. Otherwise let me know as well. Thanks.

**Comment by beckermr** [comment]
IDK if those are the only two issues.

**Comment by jbe456** [comment]
Hello @isuruf @beckermr 

We've updated our EULA: https://www.atoti.io/eula/ We now allow redistribution and we have updated paragraph 6.1 as per my last comment.

Please let us know if this comply with conda forge license requirements. If so we'll upload the new license file to the recipe. Thanks.

**Comment by beckermr** [comment]
Yeah I;m not willing to commit to these terms either

> End User may make one copy of the Software in machine readable form for backup / security purposes only, provided that the backup copy must include all proprietary notices contained on the original Software package.

Our CI+Anaconda+CDN services will end up making tons of copies for redistribution.

> 4.3. End User may not and will not allow a third party to:

> Reverse engineer, decompile, disassemble, or otherwise attempt to discover / have access to the source code or algorithms of the Software,
> Adapt, modify, disable any features and functionality of, the Software / the Results, or create derivative works based on the Software/ the Results,
> Except as provided in section 3.8 above, rent, lease, lend, sublicense or send the Software / Results or make available the Software or its functionalities to any other End User aside from Viewers,
> Remove, modify, or conceal any product identification, copyright, proprietary, intellectual property notices or other marks on or within the Software,
> Use the Software to develop a product which is competitive with any ATOTI product or services offerings."
 
We cannot control what a third-party might do with the package. 

> 6.1. The Software can be used  for a period of 64 (sixty-four) days from the date of the Software’s release (the “Term”). At the end of the Term, End-User must upgrade to a more recent version of the Software. In the absence of a Software’s release before the end of the Term, the Term will be extended until a new version is released.

We cannot force users to upgrade and we cannot be responsible for upgrading the feedstock.

**Comment by beckermr** [comment]
I am sorry about not seeing these other things sooner. I don't want to send you all on a goose chase to get this right. We need to be able to distribute the software you ship in the package to the public without any further requirements beyond the standard ones in a BSD-like license. That doesn't mean you need to use BSD (for example many packages have a non-commercial use-only requirement), but we cannot be held responsible for what users of the software conda-forge ships do with it. We ship stuff to the public.

**Comment by isuruf** [comment]
Agree with @beckermr. What you can also do is to give conda-forge an exception and allow unlimited redistribution.

cc @conda-forge/core

**Comment by jbe456** [comment]
Thanks guys for your quick replies. I understand your concerns and it was never our intention to hold conda forge responsible. We'll update the EULA and come back to you.

Meanwhile, if you have any online documentation about conda forge recipe license requirements that would be helpful! Thanks.

**Comment by jbe456** [comment]
Hello @isuruf @beckermr 

We've updated [our EULA](https://www.atoti.io/eula/) with the following changes:
- we've removed the paragraph you quoted above under section 3:
  > End User may make one copy of the Software in machine readable form for backup / security purposes only, provided that the backup copy must include all proprietary notices contained on the original Software package.
- we've added section 9.4:
  > When the Software is made available to End Users on third party platforms, these platforms cannot be held liable in any way for the failure of End Users to comply with the content of this EULA. In particular, ATOTI will be responsible for making available a new version of the Software when the previous one expires. Third party platforms are https://conda-forge.org/ and https://pypi.org/.

Please let us know if this version of the EULA now complies with conda forge license requirements. Thanks

**Comment by isuruf** [comment]
cc @ocefpaf, @jakirkham - a question to NumFOCUS legal.

**Comment by ocefpaf** [comment]
> cc @ocefpaf, @jakirkham - a question to NumFOCUS legal.

@isuruf I'll read the whole thread and try to compile an e-mail to NumFOCUS legal. Usually, just sending the GH issue/PR like, this won't help much. However, as you know my time is quite limited at the moment and this compilation may take a while to happen. If someone that is following this thread closer wants to do this summary and send it to me I will gladly forward that to NumFOCUS legal as soon as possible.

**Comment by jbe456** [comment]
Hello! do you have any updates on this topic? were you able to contact NumFOCUS legal? Thanks

**Comment by fabiencelier** [comment]
:cry:

**Pull request closed without merging**